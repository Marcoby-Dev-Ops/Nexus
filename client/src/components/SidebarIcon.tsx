import React from 'react';
import PropTypes from 'prop-types';

/**
 * @name SidebarIcon
 * @description Wrapper for sidebar nav icons for consistent sizing/alignment.
 * @param {object} props
 * @param {React.ReactNode} props.children - Icon element.
 * @param {string} [props.className] - Additional class names.
 */
interface SidebarIconProps {
  children: React.ReactNode;
  className?: string;
}

const SidebarIcon: React.FC<SidebarIconProps> = ({ children, className }) => (
  <span className={`inline-flex items-center justify-center w-5 h-5 ${className ?? ''}`}>{children}</span>
);

SidebarIcon.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
};

export default SidebarIcon; 