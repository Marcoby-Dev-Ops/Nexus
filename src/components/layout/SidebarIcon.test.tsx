/**
 * @file SidebarIcon.test.tsx
 * @description Unit and snapshot tests for the SidebarIcon component.
 */
import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import SidebarIcon from './SidebarIcon';

describe('SidebarIcon', () => {
  it('renders children', () => {
    render(<SidebarIcon><span>🏠</span></SidebarIcon>);
    expect(screen.getByText('🏠')).toBeInTheDocument();
  });

  it('applies custom className', () => {
    const { container } = render(
      <SidebarIcon className="test-class"><span>🏠</span></SidebarIcon>
    );
    expect(container.firstChild).toHaveClass('test-class');
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<SidebarIcon><span>🏠</span></SidebarIcon>);
    expect(asFragment()).toMatchSnapshot();
  });
}); 