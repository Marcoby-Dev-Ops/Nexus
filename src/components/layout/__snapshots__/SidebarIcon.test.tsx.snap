// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarIcon matches snapshot 1`] = `
<DocumentFragment>
  <span
    class="inline-flex items-center justify-center w-5 h-5 "
  >
    <span>
      🏠
    </span>
  </span>
</DocumentFragment>
`;
